### ISSUE TITLE
Resuming a .licht project with Undistort on re-undistorts already-undistorted cameras

### ISSUE THREAD

**Issue by JLMVFX** [MrNeRF/LichtFeld-Studio#2034]
**v0.5.3-505 (94957666), Windows 11, RTX PRO 6000 Blackwell.**

Train with **Undistort** enabled, save the project, close LFS, reopen the project, resume: quality drops within ~100 iterations and never recovers. Loss steps from 0.114 to ~0.155 and stays there; renders are visibly smeared.

### What the log shows

The first run undistorted 1,736 inputs, all `8192x5464` (the raw JPEGs).

On resume, the undistort **inputs** are the first run's **outputs**, count for count:

```
first run   Undistort: 8192x5464 -> 8053x5371   (x20)
            Undistort: 8192x5464 -> 8056x5373   (x19)
            Undistort: 8192x5464 -> 8050x5368   (x17)

on resume   Undistort: 8053x5371 -> 8489x5662
            Undistort: 8056x5373 -> ...
            Undistort: 8050x5368 -> ...
```

The GT pixels on disk are unchanged at 8192x5464, so after the second warp every training view is misaligned with the model.

### Cause

`Camera::prepare_undistortion()` (`src/core/camera.cpp:873`) rewrites `_focal_x/_focal_y/_center_x/_center_y/_camera_width/_camera_height` with the rectified values but keeps the distortion coefficients, and its re-entry guard `_undistort_prepared` is runtime-only:

```cpp
void Camera::prepare_undistortion(float blank_pixels) {
    if (_undistort_prepared)
        return;
    precompute_undistortion(blank_pixels);
    ...
    _camera_width  = _undistort_params.dst_width;
    _camera_height = _undistort_params.dst_height;
    _undistort_prepared = true;
}
```

The project persists the **mutated** intrinsics. On hydration `_undistort_prepared` is false again while `has_distortion()` is still true, so the trainer's "Apply undistortion to camera intrinsics" loop runs a second time on already-rectified cameras.

### Scope

* Only the cold `.licht` open + resume path is affected.
* In-process training snapshot restores (no dataset reload) are fine.
* I checked the rest of the restore path and it is correct: Adam moments / step counts / per-parameter LRs (`adopt_checkpoint_state` moves them, `reserve_capacity` preserves the live prefix), the MRNF position LR (decayed value serialized at checkpoint version 3 and re-applied), the scheduler, PPISP, the bilateral grid, and SH degree all restore properly. This is purely the camera records.

### Corroborating signal

Loss jitter (mean step-to-step |Δloss| relative to level) **fell** after the resume, 0.033 → 0.020, while the level rose. That is the signature of the targets moving, not of an unstable optimizer or a learning rate reset.

### Suggested fix

Either persist `_undistort_prepared` in the project, or make `prepare_undistortion()` idempotent with a `dst == current` check, or save the original pre-rectification intrinsics alongside the mutated ones.
